src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private int size;
+    private T[] elementsData;
+
+    public ArrayList() {
+        elementsData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            resize();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            trowException();
+        }
+        if (size == elementsData.length - 1) {
+            resize();
+        }
+        System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+        elementsData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] data = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            data[i] = list.get(i);
+        }
+        if (elementsData.length - size < list.size()) {
+            resize();
```

Comment:
`resize()` increases the size by multiplying to 1.5, what will happen if we try to addAll from the  `list` that has size  larger e.g. by 2 times than `elementsData` size?